compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -99,6 +299,14 @@ EncodingAttr::verify(function_ref<mlir::InFlightDiagnostic()> emitError,
       }
     }
   }
+  if (iterationSizesAttr) {
+    EncodingOpType opType = opTypeAttr.getValue();
+    if (opType == EncodingOpType::matmul && iterationSizesAttr.size() != 3) {
```

Comment:
I think `!= 3` does not work when it is a batch_matmul? I don't expect the logic is about `opType`. It should capture the iteration domain sizes. In linalg concept, they are values from [getStaticLoopRanges](https://github.com/llvm/llvm-project/blob/b393ca6026ffbd5621edb00503dd39e9eec97a1a/mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td#L804-L821).

I think we verify that if indexing maps are present, and whether the size matches the number of dimension of one map. Does it make sense?

Reply:
Yes, I haven't seen any example with 4 dimensions, even in the batch matmul case we only propagate 3 round_dims_to/iteration_sizes dimensions (M, N, K). I have thinking whether we should support batch size as well, but as a follow up. 

In any case, comparing with the number of dimensions in the map makes sense here.